### PR TITLE
[Graphbolt] Adapt negative sampler input

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
Change negative sampler input ot `LinkPredictionBlock`.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/negative_sampler.py [hunk 1]
```diff
@@ -38,32 +37,33 @@ def __init__(
...
         """
         Generate a mix of positive and negative samples.
 
         Parameters
         ----------
-        node_pairs : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
-            A tuple of tensors or a dictionary represents source-destination
-            node pairs of positive edges, where positive means the edge must
-            exist in the graph.
+        LinkPredictionBlock : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:46]:
```suggestion
        data : LinkPredictionBlock
```

File: python/dgl/graphbolt/negative_sampler.py [hunk 2]
```diff
@@ -38,32 +37,33 @@ def __init__(
...
-
-        data = LinkPredictionBlock(node_pair=node_pairs)
-        if isinstance(node_pairs, Mapping):
-            for etype, pos_pairs in node_pairs.items():
-                self._collate(
-                    data, self._sample_with_etype(pos_pairs, etype), etype
-                )
-        else:
-            self._collate(data, self._sample_with_etype(node_pairs))
+        if data.negative_head is None and data.negative_tail is None:
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:58]: What if the label exists?

Comment (frozenbugs) [reply python/dgl/graphbolt/negative_sampler.py:58]: I would suggest not checking here, if user call negative_sampler with node_pair, it is ok to override existing data.negative_head and data.negative_tail.

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/negative_sampler.py (hunk 2 above)
Comment (peizhou001) [reply python/dgl/graphbolt/negative_sampler.py:58]: Remove check for fix.

File: python/dgl/graphbolt/negative_sampler.py [hunk 3]
```diff
@@ -38,25 +37,26 @@ def __init__(
...
         """
         Generate a mix of positive and negative samples.
 
         Parameters
         ----------
-        node_pairs : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
-            A tuple of tensors or a dictionary represents source-destination
-            node pairs of positive edges, where positive means the edge must
-            exist in the graph.
+        LinkPredictionBlock : LinkPredictionBlock
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:46]:
```suggestion
        data : LinkPredictionBlock
```

**Pull request merged**